src/components/Table/GridTable.tsx

```diff
@@ -1108,129 +1126,146 @@ function GridRow<R extends Kinded, S>(props: GridRowProps<R, S>): ReactElement {
+                      ).$
+                    : {}),
+                }
+              : {};
+
+          const cellId = `${row.kind}_${row.id}_${column.name}`;
+          const applyCellHighlight = column.name && !isHeader && !isTotals && style.cellHighlight;
+          const isCellActive = rowState.activeCellId === cellId;
+
+          // Note that it seems expensive to calc a per-cell class name/CSS-in-JS output,
+          // vs. setting global/table-wide CSS like `style.cellCss` on the root grid div with
+          // a few descendent selectors. However, that approach means the root grid-applied
+          // CSS has a high-specificity and so its harder for per-page/per-cell business logic
+          // to override it. So, we just calc the combined table-wide+per-cell-overridden CSS here,
+          // in a very CSS-in-JS idiomatic manner.
+          //
+          // In practice we've not seen any performance issues with this from our "large but
+          // not Google spreadsheets" tables.
+          const cellCss = {
+            // Adding display flex so we can align content within the cells
+            ...Css.df.$,
+            // Apply sticky column/cell styles
+            ...maybeStickyColumnStyles,
+            // Apply any static/all-cell styling
+            ...style.cellCss,
+            // Then override with first/last cell styling
+            ...getFirstOrLastCellCss(style, columnIndex, columns),
+            // Then override with per-cell/per-row justification/indentation
+            ...justificationCss,
+            ...getIndentationCss(style, rowStyle, columnIndex, maybeContent),
+            // Then apply any header-specific override
+            ...(isHeader && style.headerCellCss),
+            // Then apply any totals-specific override
+            ...(isTotals && style.totalsCellCss),
+            // Or level-specific styling
+            ...(!isHeader && !isTotals && !!style.levels && style.levels[level]?.cellCss),
+            // Level specific styling for the first content column
+            ...(applyFirstContentColumnStyles && !!style.levels && style.levels[level]?.firstContentColumn),
+            // The specific cell's css (if any from GridCellContent)
+            ...rowStyleCellCss,
+            // Apply active row styling for non-nested card styles.
+            ...(style.nestedCards === undefined && isActive
+              ? Css.bgColor(style.activeBgColor ?? Palette.LightBlue50).$
+              : {}),
+            // Add any cell specific style overrides
+            ...(isGridCellContent(maybeContent) && maybeContent.typeScale ? Css[maybeContent.typeScale].$ : {}),
+            // And any cell specific css
+            ...(isGridCellContent(maybeContent) && maybeContent.css ? maybeContent.css : {}),
+            // Apply cell highlight styles to active cell and hover
+            ...(applyCellHighlight
+              ? isCellActive
+                ? Css.bw1.bsSolid.bLightBlue900.br4.$
```

Comment:
**question:** Does this cause all the cells to shift when we add in this border? This could impact how we define our column widths and potentially cause horizontal scrollbars when a table is already the full width of the page 🤔. It might be fine since we are using `box-sizing: border-box`, but it might cause the content within the cell to shift around. We might want to add a transparent border to every cell to prevent any shifting.